src/server/rest/v1/service/SiteService.ts

```diff
@@ -96,12 +93,36 @@ export default class SiteService {
     const site = await UtilsService.checkAndGetSiteAuthorization(
       req.tenant, req.user, filteredRequest.siteID, Action.READ, action);
```

Comment:
I thought we only need the Site as other authorizations are based on this site
1 - get the site if we have at least the right to read it
2 - get the site users auth for this site

For example a Basic user can read site but it doesn't mean he can assign users to it. He has no rights on siteUsers so it will throw an error at the checkAndGetSiteUsersAuthorization time
we can get site with the read auth for basic user but after that when we try to check dynamic auth for assignment it says `Role 'basic' is not authorized to perform 'Unassign' on 'SiteUser'`

Reply:
Yes so if instead of using Action.READ, you would use for example Action.ADD_USERS_TO_SITE (and put the action in authdefinition for Entity.SITE) you would block the basic user at this line